org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/AbstractCacheBasedTest.java

```diff
@@ -37,7 +39,13 @@ public abstract class AbstractCacheBasedTest {
 
 	static {
 		try {
-			parentDir = Files.createTempDirectory("lemminx");
+			String cacheInRepoDir = System.getProperty("cacheInRepoDir");
+			if (StringUtils.isEmpty(cacheInRepoDir)) {
```

Comment:
With the example in the build file one would think `-DcacheInRepoDir=false` should disable it, but it doesn't. Maybe it's a silly thing to expect, but I would make the check simpler. Just test if the property is not null to activate it, and otherwise use the regular behaviour. This way the build file can be activated with just `-DcacheInRepoDir` (defining it makes it non-null).